addons/mail/controllers/thread.py

```diff
@@ -208,8 +208,17 @@ def mail_message_update_content(self, message_id, body, attachment_ids, attachme
         # sudo: mail.message - access is checked in _get_with_access and _can_edit_message
         message = message.sudo()
         body = Markup(body) if body else body  # may contain HTML such as @mentions
-        guest.env[message.model].browse([message.res_id])._message_update_content(
-            message, body=body, attachment_ids=attachment_ids, partner_ids=partner_ids
+        thread = guest.env[message.model].browse([message.res_id])
+        thread._message_update_content(
+            message,
+            body=body,
+            attachment_ids=attachment_ids,
+            partner_ids=partner_ids,
+            **{
+                key: value
+                for key, value in kwargs.items()
+                if key in thread._get_allowed_message_update_params()
+            }
```

Comment:
It feels strange to let `body`, `attachment_ids`, and `partner_ids` pass through without them being in `_get_allowed_message_update_params`.

Especially when `_get_allowed_message_post_params` has all the params listed.